attestation/server-custom-mock/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use p384::ecdsa::SigningKey;
+use sec1::DecodeEcPrivateKey;
+use sha2::Digest;
+
+static ROOT_CERT: &'static [u8; 404] = include_bytes!("./root.crt");
+static LEAF_CERT: &'static [u8; 466] = include_bytes!("./leaf.crt");
+static LEAF_KEY: &'static [u8; 167] = include_bytes!("./leaf.key");
+
+pub fn get_attestation_doc(
+    public_key: Option<&[u8]>,
+    user_data: Option<&[u8]>,
+    nonce: Option<&[u8]>,
+) -> Result<Vec<u8>, String> {
+    if public_key.map_or(0, <[u8]>::len) > 65535 {
+        return Err("public key is too long, maximum of 65535".into());
+    }
+
+    if user_data.map_or(0, <[u8]>::len) > 65535 {
+        return Err("user_data is too long, maximum of 65535".into());
+    }
+
+    if nonce.map_or(0, <[u8]>::len) > 65535 {
+        return Err("nonce is too long, maximum of 65535".into());
+    }
+
+    let timestamp_ms = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_millis();
+
+    // compute sizes
```

Comment:
elaborate? didn't get you

Reply:
declare another function that accepts user inputs and generate the attestation doc. Interface similar to nsm, since we are mocking it. So that structure of get_attestation_doc looks similar to original attestation-custom-server.

Reply:
NSM api is too complicated to mock directly IMO. It has its own Request/Response structs and extra dependencies like `serde_bytes`. The most common interface between the two is likely the one we already have at the `get_attestation_doc` level. At least I don't see much more that they have in common...